src/hosts.rs

```diff
@@ -145,19 +199,28 @@ impl Host {
     /// # Returns
     ///
     /// true if the host is ready, false otherwise.
-    pub fn check_readyset_is_ready(&mut self) -> Result<bool, mysql::Error> {
+    pub fn check_readyset_is_ready(&mut self) -> Result<ProxyStatus, mysql::Error> {
         match &mut self.conn {
             Some(conn) => {
                 let result = conn.query("SHOW READYSET STATUS");
                 match result {
                     Ok(rows) => {
                         let rows: Vec<(String, String)> = rows;
                         for (field, value) in rows {
-                            if field == "Snapshot Status" {
-                                return Ok(value == "Completed");
+                            if field == "Snapshot Status" && value == "Completed" {
+                                self.readyset_status = ReadysetStatus::Online;
+                                return Ok(ProxyStatus::Online);
+                            } else if field == "Snapshot Status" && value == "In Progress" {
+                                self.readyset_status = ReadysetStatus::SnapshotInProgress;
+                                return Ok(ProxyStatus::Shunned);
```

Comment:
Alright, so we'll leave these here for backwards compatibility for a while?

Reply:
Yes, people that have multiple readysets in the same proxysql, do a rolling upgrade. The scheduler will for sometime be monitoring both versions of SHOW READYSET STATUS;